vault/resource_terraform_cloud_secret_backend.go

```diff
@@ -0,0 +1,246 @@
+	if err != nil {
+		return fmt.Errorf("Error reading mount %q: %s", path, err)
+	}
+
+	// path can have a trailing slash, but doesn't need to have one
+	// this standardises on having a trailing slash, which is how the
+	// API always responds.
+	mount, ok := mounts[strings.Trim(path, "/")+"/"]
+	if !ok {
+		log.Printf("[WARN] Mount %q not found, removing from state.", path)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("path", path)
+	d.Set("description", mount.Description)
+	d.Set("default_lease_ttl_seconds", mount.Config.DefaultLeaseTTL)
+	d.Set("max_lease_ttl_seconds", mount.Config.MaxLeaseTTL)
+
+	log.Printf("[DEBUG] Reading %s from Vault", configPath)
+	secret, err := client.Logical().Read(configPath)
+	if err != nil {
+		return fmt.Errorf("error reading from Vault: %s", err)
+	}
+
+	log.Printf("[DEBUG] secret: %#v", secret)
+
+	// token, sadly, we can't read out
+	// the API doesn't support it
+	// So... if it drifts, it drift.
+	d.Set("address", secret.Data["address"].(string))
+	d.Set("base_path", secret.Data["base_path"].(string))
+
+	return nil
+}
+
+func terraformCloudSecretBackendUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	path := d.Id()
+	configPath := terraformCloudSecretBackendConfigPath(path)
+
+	d.Partial(true)
+
+	if d.HasChange("default_lease_ttl_seconds") || d.HasChange("max_lease_ttl_seconds") {
+		config := api.MountConfigInput{
+			DefaultLeaseTTL: fmt.Sprintf("%ds", d.Get("default_lease_ttl_seconds")),
+			MaxLeaseTTL:     fmt.Sprintf("%ds", d.Get("max_lease_ttl_seconds")),
+		}
+
+		log.Printf("[DEBUG] Updating lease TTLs for %q", path)
+		if err := client.Sys().TuneMount(path, config); err != nil {
+			return fmt.Errorf("Error updating mount TTLs for %q: %s", path, err)
+		}
+
+		d.SetPartial("default_lease_ttl_seconds")
+		d.SetPartial("max_lease_ttl_seconds")
+	}
+	if d.HasChange("address") || d.HasChange("token") || d.HasChange("base_path") {
+		log.Printf("[DEBUG] Updating Terraform Cloud configuration at %q", configPath)
+		data := map[string]interface{}{
+			"address":   d.Get("address").(string),
+			"token":     d.Get("token").(string),
+			"base_path": d.Get("base_path").(string),
+		}
+		if _, err := client.Logical().Write(configPath, data); err != nil {
+			return fmt.Errorf("Error configuring Terraform Cloud configuration for %q: %s", path, err)
+		}
+		log.Printf("[DEBUG] Updated Terraform Cloud configuration at %q", configPath)
+		d.SetPartial("address")
+		d.SetPartial("token")
+		d.SetPartial("base_path")
+	}
+	d.Partial(false)
+	return terraformCloudSecretBackendRead(d, meta)
+}
+
+func terraformCloudSecretBackendDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	path := d.Id()
+
+	log.Printf("[DEBUG] Unmounting Terraform Cloud backend %q", path)
+	err := client.Sys().Unmount(path)
+	if err != nil {
+		return fmt.Errorf("Error unmounting Terraform Cloud backend from %q: %s", path, err)
```

Comment:
We should check for a 404 style error here; if so, remove from state and log that it wasn't found. Example: https://github.com/hashicorp/terraform-provider-vault/blob/3a348eac1c3cacbefbee3166e18f03f9c59824b5/vault/resource_ad_secret_backend.go#L645

Reply:
I suppose that's assuming this isn't already accounted for because of the `Exists` method provided, in which case, we can ignore this suggestion :D

Reply:
Suggestion ignored. :)